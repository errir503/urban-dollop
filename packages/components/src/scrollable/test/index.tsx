/**
 * External dependencies
 */
import { render, screen } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { Scrollable } from '../index';

describe( 'props', () => {
	beforeEach( () => {
		render(
			<Scrollable data-testid="scrollable">
				WordPress.org - Code is Poetry
			</Scrollable>
		);
	} );

	test( 'should render correctly', () => {
		expect( screen.getByTestId( 'scrollable' ) ).toMatchSnapshot();
	} );

	test( 'should render smoothScroll', () => {
		render(
			<Scrollable smoothScroll data-testid="smooth-scrollable">
				WordPress.org - Code is Poetry
			</Scrollable>
		);
		expect(
			screen.getByTestId( 'smooth-scrollable' )
		).toMatchStyleDiffSnapshot( screen.getByTestId( 'scrollable' ) );
	} );
} );
