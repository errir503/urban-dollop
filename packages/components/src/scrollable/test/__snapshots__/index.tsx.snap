// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props should render correctly 1`] = `
.emotion-0 {
  height: 100%;
  overflow-x: hidden;
  overflow-y: auto;
}

@media only screen and ( min-device-width: 40em ) {
  .emotion-0::-webkit-scrollbar {
    height: 12px;
    width: 12px;
  }

  .emotion-0::-webkit-scrollbar-track {
    background-color: transparent;
  }

  .emotion-0::-webkit-scrollbar-track {
    background: rgba(0, 0, 0, 0.04);
    border-radius: 8px;
  }

  .emotion-0::-webkit-scrollbar-thumb {
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    background-color: rgba(0, 0, 0, 0.2);
    border: 2px solid rgba( 0, 0, 0, 0 );
    border-radius: 7px;
  }

  .emotion-0:hover::-webkit-scrollbar-thumb {
    background-color: rgba(0, 0, 0, 0.5);
  }
}

<div
  class="components-scrollable emotion-0 emotion-1"
  data-testid="scrollable"
  data-wp-c16t="true"
  data-wp-component="Scrollable"
>
  WordPress.org - Code is Poetry
</div>
`;

exports[`props should render smoothScroll 1`] = `
Snapshot Diff:
- Received styles
+ Base styles

  Array [
    Object {
      "height": "100%",
      "overflow-x": "hidden",
      "overflow-y": "auto",
-     "scroll-behavior": "smooth",
    },
  ]
`;
